### PR TITLE
checking if usecli environment variable is set

### LINKED ISSUE #334: Able to set environment variable to set use_cli config
To be able to remove hard-coded `use_cli = {true, false}` config, provider code should check if an environment variable is exists, like `POWER_PLATFORM_USE_CLI`.

Similar need as #329

### PR DESCRIPTION
**Pull request by polatengin**
To be able to remove hard-coded `use_cli = {true, false}` config from provider section, provider should be able to determine if a specific environment variable is set.

Before;

```terraform
provider "powerplatform" {
  use_cli = true
}
```

After _(with `POWER_PLATFORM_USE_CLI` environment variable is set)_;

```terraform
provider "powerplatform" {
}
```

closes #334

### CODE REVIEW

**Review by eduardodfmex — APPROVED** [review]

**Comment by ianjensenisme** [comment]
Why does the environment config trump the provider config? This is more of a "food for thought" question, as I've also added variable handling that uses this priority order, so not worth changing, but I wonder if we've gotten it backwards throughout the provider. I'm probably just not thinking of an obvious reason to do things in that priority order. Anywho, LGTM.

**Review by ianjensenisme — DISMISSED** [review]
Why does the environment config trump the provider config? This is more of a "food for thought" question, as I've also added variable handling that uses this priority order, so not worth changing, but I wonder if we've gotten it backwards throughout the provider. I'm probably just not thinking of an obvious reason to do things in that priority order. Anywho, LGTM.

File: internal/powerplatform/provider.go [hunk 1]
```diff
@@ -181,6 +181,14 @@ func (p *PowerPlatformProvider) Configure(ctx context.Context, req provider.Conf
 		clientSecret = config.ClientSecret.ValueString()
 	}
 
+	useCli := false
+	_, envUseCliExists := os.LookupEnv("POWER_PLATFORM_USE_CLI")
+	if config.UseCli.IsNull() {
+		useCli = envUseCliExists
+	} else {
+		useCli = config.UseCli.ValueBool()
+	}
```
Comment (mattdot) [inline internal/powerplatform/provider.go:190]:
```suggestion
	_, envUseCli := os.LookupEnv("POWER_PLATFORM_USE_CLI")
	if config.UseCli.IsNull() {
		useCli = envUseCli
	} else {
		useCli = config.UseCli.ValueBool()
	}
```

**Review by mattdot — CHANGES_REQUESTED** [review]

File: internal/powerplatform/provider.go (hunk 1 above)
Comment (polatengin) [reply internal/powerplatform/provider.go]: fair 👍 made the changes..

**Review by mawasile — APPROVED** [review]

**Review by eduardodfmex — APPROVED** [review]

**Review by mattdot — APPROVED** [review]

**Pull request merged**